### DISCUSSION TITLE
How to get the 'Content-Length' please?

### DISCUSSION THREAD

**Discussion by acycliq** [community/community#21488] [Other Feature Feedback, Questions, & Ideas]
<p>How to get the <code>content-length</code> of a response please? <code>response.headers.get('Content-Length')</code> returns <code>null</code>.<br>
See the example below, which is taken from this <a href="https://javascript.info/fetch-progress" rel="nofollow noopener">page</a></p>
<pre><code class="lang-auto">&lt;!DOCTYPE html&gt;
&lt;html lang="en"&gt;
&lt;head&gt;
    &lt;meta charset="UTF-8"&gt;
    &lt;title&gt;Title&lt;/title&gt;
&lt;/head&gt;
&lt;body&gt;

&lt;script&gt;
    // Step 1: start the fetch and obtain a reader
    const start = async function (a, b) {
        let response = await fetch('https://api.github.com/repos/javascript-tutorial/en.javascript.info/commits?per_page=100');

        const reader = response.body.getReader();

// Step 2: get total length
        const contentLength = +response.headers.get('Content-Length');

// Step 3: read the data
        let receivedLength = 0; // received that many bytes at the moment
        let chunks = []; // array of received binary chunks (comprises the body)
        while (true) {
            const {done, value} = await reader.read();

            if (done) {
                break;
            }

            chunks.push(value);
            receivedLength += value.length;

            console.log(`Received ${receivedLength} of ${contentLength}`)
        }

// Step 4: concatenate chunks into single Uint8Array
        let chunksAll = new Uint8Array(receivedLength); // (4.1)
        let position = 0;
        for (let chunk of chunks) {
            chunksAll.set(chunk, position); // (4.2)
            position += chunk.length;
        }

// Step 5: decode into a string
        let result = new TextDecoder("utf-8").decode(chunksAll);

// We're done!
        let commits = JSON.parse(result);
        alert(commits[0].author.login);
    }

    start()
&lt;/script&gt;

&lt;/body&gt;
&lt;/html&gt;
</code></pre>

**Comment by airtower-luna** [comment]
<p>The answer is right there on <a href="https://javascript.info/fetch-progress">the page you linked</a>:</p>
<blockquote>
<p>It may be absent for cross-origin requests (see chapter <a href="https://javascript.info/fetch-crossorigin">Fetch: Cross-Origin Requests</a>) and, well, technically a server doesn’t have to set it. But usually it’s at place.</p>
</blockquote>
<p>And I’d question the claim that it’s “usually” there. For static data that’s true, but dynamically generated data will pretty often be sent without a Content-Length header, using chunked encoding.</p>

**Comment by acycliq** [comment]
<p>So, how can one get the <code>Content-Length</code> then?  Or this is not possible since the server doesnt send it?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You can’t get a header the server never sent. If you need the length of<br>
the received data you can calculate that after receiving all of it. Note<br>
however that some HTTP responses may not have an end, e.g. when<br>
receiving an audio or video stream (depending on the format).</p>

**Comment by acycliq** [comment]
<p>OK, many thanks for that! I wasnt quite sure to be honest. I looks like an important piece of info to be left out</p>
